### PR TITLE
feat(open-api-gateway): add java projen project type for defining cdk infrastructure in java

### PR DESCRIPTION
**Pull request by cogwirrel**
Add `OpenApiGatewayJavaProject` to complete the set of three (typescript, python and java)!

This generates the type-safe interface for the CDK construct in Java, as well as a sample construct and lambda handler for an out of the box hello world api.

Note also I address an issue with the generated python and java clients where nx was not picking up the generated client as a project in the repo, because the monorepo adds the `package.json` file but by that point the client had already been synthesized. Addressed this by adding the `package.json` ourselves and skipping it in the monorepo project. This enables us to set up the implicit dependency between the generated python client and the python api project (and the same for java) as well :).

### CODE REVIEW

File: packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts [hunk 1]
```diff
@@ -83,6 +84,19 @@ export class GeneratedJavaClientProject extends JavaProject {
...
+    new JsonFile(this, "package.json", {
+      obj: {
+        name: this.name,
+        __pdk__: true,
+        version: options.version,
+        scripts: Object.fromEntries(
+          this.tasks.all.map((task) => [task.name, `npx projen ${task.name}`])
+        ),
+      },
+      readonly: true,
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts:98]: should we also set the marker given this is generated?

Comment (agdimech) [reply packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts:98]: same for all other langs.

File: packages/open-api-gateway/src/project/open-api-gateway-java-project.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+import { DocsProject } from "./codegen/docs-project";
+import { generateClientProjects } from "./codegen/generate";
+import { GeneratedJavaClientProject } from "./codegen/generated-java-client-project";
+import { ClientLanguage } from "./languages";
+import { getJavaSampleSource, JavaSampleCodeOptions } from "./samples/java";
+import { OpenApiSpecProject } from "./spec/open-api-spec-project";
+import { OpenApiGatewayProjectOptions } from "./types";
+
+const OPENAPI_GATEWAY_PDK_PACKAGE_NAME =
+  "software.aws.awsprototypingsdk/open-api-gateway@^0";
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/open-api-gateway-java-project.ts:30]: well need to remember to bump this when we release 1.0

File: packages/open-api-gateway/src/project/open-api-gateway-java-project.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      this.specFileName = "spec.yaml";
+    }
+    this.generatedCodeDir = options.generatedCodeDir ?? "generated";
+    this.forceGenerateCodeAndDocs = options.forceGenerateCodeAndDocs ?? false;
+    this.apiSrcDir = options.apiSrcDir ?? "api";
+
+    // Generated project should have a dependency on this project, in order to run the generation scripts
+    [
+      OPENAPI_GATEWAY_PDK_PACKAGE_NAME,
+      "software.constructs/constructs@^10",
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/open-api-gateway-java-project.ts:104]: java deps in projen are painful...

**Review by agdimech — APPROVED** [review]

File: packages/open-api-gateway/src/project/open-api-gateway-java-project.ts (hunk 3 above)
Comment (cogwirrel) [reply packages/open-api-gateway/src/project/open-api-gateway-java-project.ts:104]: Yeah it's a shame we have to specify a version :(

File: packages/open-api-gateway/src/project/open-api-gateway-java-project.ts (hunk 2 above)
Comment (cogwirrel) [reply packages/open-api-gateway/src/project/open-api-gateway-java-project.ts:30]: Yeah, annoying!

File: packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts (hunk 1 above)
Comment (cogwirrel) [reply packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts:98]: I think given that all the rest of the generated client code won't have the marker there's not much benefit in adding it for the `package.json`. Greg is actually looking into adding more general cleanup of client projects - ie when you remove a `ClientLanguage` it deletes the appropriate directory, or when you change the spec you don't end up with old dangling generated files :)

Comment (agdimech) [reply packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts:98]: Ok sounds good.

**Pull request merged**